homeassistant/components/cover/__init__.py

```diff
@@ -0,0 +1,255 @@
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_OPEN_COVER, data)
+
+
+def close_cover(hass, entity_id=None):
+    """Close all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_CLOSE_COVER, data)
+
+
+def set_cover_position(hass, position, entity_id=None):
+    """Move to specific position all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    data[ATTR_POSITION] = position
+    hass.services.call(DOMAIN, SERVICE_SET_COVER_POSITION, data)
+
+
+def stop_cover(hass, entity_id=None):
+    """Stop all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_STOP_COVER, data)
+
+
+def open_cover_tilt(hass, entity_id=None):
+    """Open all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_OPEN_COVER_TILT, data)
+
+
+def close_cover_tilt(hass, entity_id=None):
+    """Close all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_CLOSE_COVER_TILT, data)
+
+
+def set_cover_tilt_position(hass, tilt_position, entity_id=None):
+    """Move to specific tilt position all or specified cover."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    data[ATTR_TILT_POSITION] = tilt_position
+    hass.services.call(DOMAIN, SERVICE_SET_COVER_TILT_POSITION, data)
+
+
+def stop_cover_tilt(hass, entity_id=None):
+    """Stop all or specified cover tilt."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
+    hass.services.call(DOMAIN, SERVICE_STOP_COVER_TILT, data)
+
+
+def setup(hass, config):
+    """Track states and offer events for covers."""
+    component = EntityComponent(
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL, GROUP_NAME_ALL_COVERS)
+    component.setup(config)
+
+    def handle_cover_service(service):
+        """Handle calls to the cover services."""
+        target_covers = component.extract_from_service(service)
+
+        for cover in target_covers:
+            if service.service == SERVICE_OPEN_COVER:
+                cover.open_cover()
+            elif service.service == SERVICE_CLOSE_COVER:
+                cover.close_cover()
+            elif service.service == SERVICE_SET_COVER_POSITION:
+                cover.set_position_cover(service.data[ATTR_POSITION])
+            elif service.service == SERVICE_STOP_COVER:
+                cover.stop_cover()
+            elif service.service == SERVICE_OPEN_COVER_TILT:
+                cover.open_cover_tilt()
+            elif service.service == SERVICE_CLOSE_COVER_TILT:
+                cover.close_cover_tilt()
+            elif service.service == SERVICE_SET_COVER_TILT_POSITION:
+                cover.set_cover_tilt_position(service.data[ATTR_TILT_POSITION])
```

Comment:
You can clean this up by using a similar approach as in `media_player` and `light` component:
```py
SERVICE_TO_METHOD = {
    SERVICE_OPEN_COVER: {'method': 'open_cover'},
    SERVICE_CLOSE_COVER: {'method': 'close_cover'},
    SERVICE_SET_COVER_POSITION: {
        'method': 'set_position_cover',
        'schema': COVER_SET_COVER_POSITION_SCHEMA},
    SERVICE_STOP_COVER: {'method': 'stop_cover'},
    SERVICE_OPEN_COVER_TILT: {'method': 'open_cover_tilt'},
    SERVICE_CLOSE_COVER_TILT: {'method': 'close_cover_tilt'},
    SERVICE_SET_COVER_TILT_POSITION: {
        'method': 'set_cover_tilt_position',
        'schema': COVER_SET_COVER_TILT_POSITION_SCHEMA},
}
...

def setup(hass, config):
...
    
    def handle_cover_service(service):
        """Map services to methods on CoverDevice."""
        method = SERVICE_TO_METHOD.get(service.service)
        params = service.data.copy()
        params.pop(ATTR_ENTITY_ID, None)

        if method:
            for cover in component.extract_from_service(service):
                getattr(cover, method)(**params)

                if cover.should_poll:
                    cover.update_ha_state(True)

    descriptions = load_yaml_config_file(
        os.path.join(os.path.dirname(__file__), 'services.yaml'))

    for service_name in SERVICE_TO_METHOD:
        if 'schema' in SERVICE_TO_METHOD[service_name]:
            schema = SERVICE_TO_METHOD[service_name]['schema']
        else:
            schema = COVER_SERVICE_SCHEMA
        hass.services.register(DOMAIN, service_name, handle_cover_service,
                               descriptions.get(service_name), schema=schema)
```
It saves about 20 lines and makes for easier future extension.